pkg/sql/parser/sql.y

```diff
@@ -4425,6 +4421,65 @@ constraint_elem:
     }
   }
 
+
+ctas_opt_col_list:
+  '(' ctas_table_elem_list ')'
+  {
+    $$.val = $2.val
+  }
+| /* EMPTY */
+  {
+    $$.val = tree.TableDefs(nil)
+  }
+
+ctas_table_elem_list:
+  ctas_table_elem
+  {
+    $$.val = tree.TableDefs{$1.tblDef()}
+  }
+| ctas_table_elem_list ',' ctas_table_elem
+  {
+    $$.val = append($1.tblDefs(), $3.tblDef())
+  }
+
+ctas_table_elem:
+  ctas_column_def
```

Comment:
another catch from @rolandcrosby: in additional to column def (indeed, more important than being able to put a qualifier on one of the names in fact) we also want the ability to say `PRIMARY KEY(a, b)` as to define a new multi-col PK (that's where this really gets interesting in repartitioning). I think we could allow it as an alternative _elem case if we wanted, and then handle it or a qualifier. Alternatively, that could be the only syntax for specifying the PK (instead of the qualifier) which eliminated conflicts. I don't have a preference so I'll let you and @rolandcrosby figure that out

Reply:
for me, supporting multi-col PK at all is a higher priority than supporting multiple syntaxes to define PKs, but otherwise I don't have a strong opinion on whether it's worth supporting this via both inline qualifiers and the `PRIMARY KEY (a, b)` syntax.